[2.9.0] Debug mode for fleet in rancher (#1350)
rancher/rancher-docs: docs/integrations-in-rancher/fleet/overview.md

- **Temporary Workaround**: By default, user-defined secrets are not backed up in Fleet. It is necessary to recreate secrets if performing a disaster recovery restore or migration of Rancher into a fresh cluster. To modify resourceSet to include extra resources you want to backup, refer to docs [here](https://github.com/rancher/backup-restore-operator#user-flow).

> -  **Debug logging**: To enable debug logging of Fleet components, you can create a new **fleet** entry in the existing **rancher-config** ConfigMap in the **cattle-system** namespace with the value `{"debug": 1, "debugLevel": 1}`. The Fleet application is restarted after saving the ConfigMap, and debug mode is enabled.

Reviewer: @thehejik in this [comment](https://github.com/rancher/fleet/issues/2345#issuecomment-2163180716) you reference using this method to enable debug mode for 2.9. Is there a method for versions <2.9?

Author: The configmap method for configuring rancher apps seems to be present in rancher v2.7 and v2.8 as well, so it should work the same way. I checked the functionality on current v2.8-head.

This PR was made due to another default loglevel settings in Fleet 0.10/v2.9 where some messages are not visible as before in v2.8/v2.7. Anyway the debug values settings for fleet chart did not change and they are usable for any Fleet version.

> -  **Debug logging**: To enable debug logging of Fleet components, you can create a new **fleet** entry in the existing **rancher-config** ConfigMap in the **cattle-system** namespace with the value `{"debug": 1, "debugLevel": 1}`. The Fleet application is restarted after saving the ConfigMap, and debug mode is enabled.

Reviewer 2: ```suggestion
-  **Debug logging**: To enable debug logging of Fleet components, create a new **fleet** entry in the existing **rancher-config** ConfigMap in the **cattle-system** namespace with the value `{"debug": 1, "debugLevel": 1}`. The Fleet application restarts after you save the ConfigMap.
```

## Documentation

> The Fleet documentation is at https://fleet.rancher.io/.

Reviewer 2: ```suggestion
See the [official Fleet documentation](https://fleet.rancher.io/) to learn more.
```

This isn't really part of the PR but should be updated to avoid having bare linked URLs in the docset
